model-optimizer/extensions/back/ShuffleChannelPatternOptimization.py

```diff
@@ -126,3 +130,145 @@ def replace_pattern(graph: Graph, match: dict):
 
         match['reshape_1_dim']['value'] = int64_array(np.take(new_end.in_port(1).data.get_value(), [0, 3, 1, 2]))
         match['reshape_1_dim'].infer(match['reshape_1_dim'])
+
+
+class ShuffleChannelFusion(BackReplacementPattern):
+    """
+    FUSION: Reshape->Transpose->Reshape  to  ShuffleChannel
```

Comment:
Please, add a comment that the transformation fixes reshape-ability of the model.